clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -2047,6 +2109,12 @@ EmitCheckedMixedSignMultiply(CodeGenFunction &CGF, const clang::Expr *Op1,
   }
   assert(Overflow && Result && "Missing overflow or result");
 
+  if (ResultIsCap)
+    Result = ProvenanceCap
+                 ? CGF.setCapabilityIntegerValue(ProvenanceCap, Result, Loc)
+                 : CGF.getNullDerivedCapability(
+                       CGF.CGM.getTypes().ConvertType(ResultQTy), Result);
```

Comment:
setCapabilityIntegerValue() with a `null` input will call `CGF.getNullDerivedCapability`, so maybe it would be simpler to just assign a `ConstantPointerNull` to `ProvenanceCap`?